### PR TITLE
React router upgrade

### PR DESCRIPTION
**Pull request by erchaves**
## Update react-router to the latest version (React Router 4)

React Router is an important part of the react ecosystem. KYT currently uses a significantly older version of react router. In order to keep KYT up to date and compatible with future development, this RFC proposes that we update to react router 4.

For more details, please see [RFC 559](https://github.com/NYTimes/kyt/issues/559)

### CODE REVIEW

**Comment by erchaves** [comment]
Hmm, it looks like the e2e test is failing at:
`  ● starter kyts › kyt-starter-universal › should start a dev server on :3000`

I'm not sure how to debug that at the moment. When I use `kyt-cli setup --local-path` to make a copy of the updated package, it builds and runs fine for me, and the tests pass.

**Comment by delambo** [comment]
Hey, Eric - I think the easiest way to test something like this is to run the e2e tests locally from the root directory: `npm run e2e`. To make it easier, you might want to comment out the other tests in `e2e_tests/tests/*.js` or specify the test you want to run in: `e2e_tests/jest.config.json`.

We're slowly working toward RR4, as well. We haven't had a lot of time or need for it, so it will be interesting to see how this works.

**Comment by erchaves** [comment]
Hi @delambo, thanks for the response. I figured out the issue with the tests and got it fixed.  

As referenced above, here is the accompanying [RFC 559](https://github.com/NYTimes/kyt/issues/559). Here's a short summary: 
- This updates us to the latest react router, keeping us up to date with their documentation.
- It removes the need for the react-hot-loader work around.
- A possible down side is that it removes the code-splitting code. I'm not sure if that is a priority to keep included in the default package. If it is, the RFC discusses how that could possibly be remedied through a little more investigation. 

If this feels like the right approach, let me know if I can assist further. I'm hoping to use kyt on a number of projects including some open source work for museums. I can share more details about that as well if you're interested.  Thanks!

File: packages/kyt-starter-universal/starter-src/src/routes/index.js [hunk 1]
```diff
@@ -1,43 +1,58 @@
...
+  children: PropTypes.node.isRequired,
+};
+
+RedirectWithStatus.propTypes = {
+  from: PropTypes.string.isRequired,
+  to: PropTypes.string.isRequired,
+  status: PropTypes.number.isRequired,
+};
+
+export default Routes();
```
Comment (staylor) [inline packages/kyt-starter-universal/starter-src/src/routes/index.js]: does this need to be a singleton?

Comment (erchaves) [reply packages/kyt-starter-universal/starter-src/src/routes/index.js]: Good catch @staylor, thx. I've updated this.

**Comment by delambo** [comment]
Again, sorry for the late treatment. @staylor added RR4 in some of his recent work. A 1.0 release will follow, hopefully soon.

**Pull request closed without merging**